opal/mca/btl/base/btl_base_select.c

```diff
@@ -107,6 +122,16 @@ int mca_btl_base_select(bool enable_progress_threads,
                                 "select: no init function; ignoring component %s",
                                 component->btl_version.mca_component_name);
         } else {
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
+            if (strcmp(component->btl_version.mca_component_name, "openib") == 0) {
```

Comment:
@jsquyres thank you for sharing details. Actually my initial local version did something you described. Reason I did not push one was in necessity to use #ifdef inside memory/base that relates memory/linux component only. It seems that solaris does not use hooks.